File: src/main/java/com/vinyl/model/Check.java
package com.vinyl.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Check {
	private Integer checkNum;
	private Timestamp dateTime;
	private Integer salesmanTabNum;
	private Integer customerNum;
	private Short checkDiscount;
	private Integer overallSum;
	private Integer sumWithDiscount;
	private List<String> productBarcodes;

	public Check(Timestamp dateTime, Integer salesmanTabNum, Integer customerNum, List<String> productBarcodes) {
		this.dateTime = dateTime;
		this.salesmanTabNum = salesmanTabNum;
		this.customerNum = customerNum;
		this.productBarcodes = productBarcodes;
	}
}

File: src/main/resources/frontend/src/schemas/check.js
export default {
  props: {
    checkNum: 'int',
    dateTime: 'datetime',
    customerNum: {
      type: 'customer',
      isSchema: true,
      isNullable: true
    },
    salesmanTabNum: {
      type: 'salesman',
      isSchema: true
    },
    overallSum: 'int',
    checkDiscount: 'int',
    sumWithDiscount: 'int',
    productBarcodes: {
      type: 'record',
      isSchema: true,
      isArray: true
    }
  },
  key: 'checkNum',
  display: (item) => item.checkNum
}